File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int temp) {
     HashSet<Integer> hs=new HashSet<>();
     while(true){
        int s=0;
        while(temp!=0){
            int a=temp%10;
            s+=a*a;
            temp/=10;
        }
        temp=s;
        if(s==1){
            return true;
        }
        if(hs.contains(temp)){
            return false;
        }
        hs.add(temp);
     }   
    }
}

File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        if n==1:
            return True
        keep=set()
        ele=str(n)
        while ele not in keep:
            keep.add(ele)
            summ=0
            for digit in ele:
                digit=int(digit)
                summ+=digit**2
            if summ==1: return True
            ele=str(summ)
        return False